src/mlpack/methods/ann/layer/sequential_impl.hpp

```diff
@@ -128,6 +145,11 @@ void Sequential<InputDataType, OutputDataType, CustomLayers...>::Backward(
   }
 
   g = boost::apply_visitor(deltaVisitor, network.front());
+
+  if (residual)
+  {
+    g += gy;
```

Comment:
Why this step?

Reply:
That's just the backward step of a residual block. As y = f(x) + x in a residual block instead of just y = f(x), we need to add the error `gy` to the gradient `g` for a correct backward pass.
dy/dx = d(f(x))/dx + 1
g = gy * dy/dx
g = gy * d(f(x))/dx + gy